frontend/src/employee-frontend/components/holiday-term-periods/TermsSection.tsx

```diff
@@ -26,12 +34,63 @@ export default React.memo(function TermsSection() {
   const { i18n } = useTranslation()
   const navigate = useNavigate()
 
+  const [termToEdit, setTermToEdit] = useState<UUID>()
+  const [showEditModal, setShowEditModal] = useState(false)
+
+  const [termToDelete, setTermToDelete] = useState<UUID>()
+  const [showDeletionModal, setShowDeletionModal] = useState(false)
+
   const preschoolTerms = useQueryResult(preschoolTermsQuery())
 
+  const { mutateAsync: deletePreschoolTerm } = useMutationResult(
+    deletePreschoolTermMutation
+  )
+
+  const onDeletePreschoolTerm = useCallback(
+    (termId: UUID) => {
+      setTermToDelete(termId)
+      setShowDeletionModal(true)
+    },
+    [setTermToDelete, setShowDeletionModal]
+  )
+
+  const deletePreschoolTermHandle = useCallback(
+    () => (termToDelete ? deletePreschoolTerm(termToDelete) : Promise.reject()),
+    [termToDelete, deletePreschoolTerm]
+  )
+
+  const onCloseDeletionModal = useCallback(() => {
+    setTermToDelete(undefined)
+    setShowDeletionModal(false)
+  }, [setTermToDelete, setShowDeletionModal])
+
+  const onEditTermHandle = useCallback(
+    (term: PreschoolTerm) => {
+      if (term.finnishPreschool.start.isBefore(LocalDate.todayInSystemTz())) {
+        setTermToEdit(term.id)
+        setShowEditModal(true)
```

Comment:
consider if both states are needed or can the modal be shown whenever termToEdit is not null

Reply:
Very true, I removed showEditModal state completely.